### DISCUSSION TITLE
readAt and multiple initialization impact of prisma client.

### DISCUSSION THREAD

**Discussion by dotsinspace** [prisma/prisma#23751] [Q&A] [answered]
### Question

I have two questions.

1. is there any possibility to use something like readAt directive just like we have updatedAt for automatic update of row with latest timestamp.

Usage case -> Seeing if user is active or not based on last query read timestamp.

2. I am using bree can i initialize prisma client multiple times without impacting the memory.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @dotsinspace 👋 

> Is there any possibility to use something like readAt directive just like we have updatedAt for automatic update of row with latest timestamp

Prisma currently does not provide a built-in directive like `@readAt` similar to `@updatedAt` for automatically updating a timestamp when a row is read. For your use case, you might need to manually update a `readAt` field each time a user reads a record.

> I am using bree can i initialize prisma client multiple times without impacting the memory.

Prisma Client is designed to be initialized once in your application and then imported wherever it's needed. It's a singleton, meaning that after you instantiate it once, the same instance is returned every time you import it again in your code. This is the recommended way to use Prisma Client in your application to prevent multiple instances which can lead to high memory usage. See this [section](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/instantiate-prisma-client#the-number-of-prismaclient-instances-matters) of the documentation.
